### ISSUE TITLE
Material doesn't have a texture property '_MainTex'

### ISSUE THREAD

**Issue by boxseto** [ColinLeung-NiloCat/UnityURPToonLitShaderExample#7]
Hi, 

I am very new to both unity and shader, and I come across your brilliant project.
I just tried to pull your code and import to a new project I loaded together with Unity-chan. I tried to change her body material to this project, but everything goes purple and the following error was thrown

`Material 'body' with Shader 'SimpleURPToonLitExample(With Outline)' doesn't have a texture property '_MainTex' UnityEngine.GUIUtility:ProcessEvent (int,intptr,bool&)`

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/35777054/106375847-ba980280-63ca-11eb-8a1f-c3ed80217657.png">

I have tried to 
  1. add the original picture to `_BaseMap (Albedo)`
  2. change the variable `_BaseMap` to `_MainTex` in `SimpleURPToonLitOutlineExample.shader`
but both does not work...

If it helps, I am using Unity version 2020.1.f11.

I am very new to these tools and concepts, I am sorry if I asked a stupid question.

**Comment by Kadae** [comment]
Hi, you can try to add this to the shader...

`_MainTex("Main Texture", 2D) = "SimpleURPToonLitExample(With Outline)" {}`

I have similar issues on another shaders, but this is a hotfix, lol.

**Comment by boxseto** [comment]
I tried but it seems not working...
Do you mean adding the line in `properties` section of `SimpleURPToonLitOutlineExample.shader` file?
I tried to add there, plus adding/removing the code for `_BaseMap`, but still the texture does not render.

Can I ask if your fix is trying to "add a new texture to the shader" or "replace the texture of the _BaseMap texture"?

**Comment by Kadae** [comment]
I just added this to Properties, but this helps in another shader, not in this.
Important: the name must be the same as the Shader's name (first line of code).

`_MainTex("Main Texture", 2D) = "** SHADER NAME **" {}`

If this not works, so... I don't know, sorry.

**Comment by boxseto** [comment]
OK, so after some testing and poking around, I found that the problem is I used the wrong pipeline...

Make sure you choose "Universal Render Pipeline" when creating a project: 
<img width="908" alt="image" src="https://user-images.githubusercontent.com/35777054/106385535-be984480-640b-11eb-8d8a-72a45d62645a.png">

I have chosen "3d", but it does not render correctly even after I imported URP in packet manager (not sure why, so you may need to search on swapping pipeline if you are upgrading it).

For those tries to use this with Unity Chan (For Absolute Beginners like me):
  1. Open new project with URP selected
  2. Install Unity chan and this project
  3. go to `/Assets/unity-chan!/Unity-chan! Model/Art/Models` and put the model to scene
  4. go to `/Assets/unity-chan!/Unity-chan! Model/Art/Materials`, the preview should be all wrong; this indicate that the pipeline of original Unity-Chan is using a different rendering pipeline
  5. for each of the material, change its shader to the one in this project

In the end something like this should be seen:
<img width="1437" alt="image" src="https://user-images.githubusercontent.com/35777054/106385821-1aaf9880-640d-11eb-86e6-d7b690460a7d.png">

It does not look good but at least it renders. From here I can try to poke around with the project code. Thanks!

TL;DR: Purple is wrong pipeline, Are you sure you are using URP?
